tensorflow/core/kernels/whole_file_read_ops.cc

```diff
@@ -119,4 +119,34 @@ class ReadFileOp : public OpKernel {
 
 REGISTER_KERNEL_BUILDER(Name("ReadFile").Device(DEVICE_CPU), ReadFileOp);
 
+class WriteFileOp : public OpKernel {
+  public:
+    using OpKernel::OpKernel;
+    void Compute(OpKernelContext* context) override {
+	    const Tensor* filename_input;
+	    const Tensor* contents_input;
+	    Tensor* output = nullptr;
+	    OP_REQUIRES_OK(context, context->input("filename", &filename_input));
+	    OP_REQUIRES_OK(context, context->input("contents", &contents_input));
+	    OP_REQUIRES(context, TensorShapeUtils::IsScalar(filename_input->shape()),
+			    errors::InvalidArgument(
+				    "Input filename tensor must be scalar, but had shape: ",
+				    filename_input->shape().DebugString()));
+	    OP_REQUIRES(context, TensorShapeUtils::IsScalar(contents_input->shape()),
+			    errors::InvalidArgument(
+				    "Contents tensor must be scalar, but had shape: ",
+				    contents_input->shape().DebugString()));
+	    OP_REQUIRES_OK(context, context->allocate_output("output", TensorShape({}),
+				    &output));
+	    if(WriteStringToFile(context->env(),
+				    filename_input->scalar<string>()(),
+				    contents_input->scalar<string>()()).ok()) {
+	    	output->scalar<int>()(0) = 0;
+	    } else {
+	    	output->scalar<int>()(0) = -1;
```

Comment:
Thanks,
I have a question; If we get rid of output, what would be the appropriate update we set to the op `SetShapeFn` in `tensorflow/core/kernels/io_ops.cc`, and also for the shape registration in 
`tensorflow/python/ops/io_ops.py` ?